Runtime/JSArray.Enumerator.cs

```diff
@@ -8,17 +8,17 @@ public partial struct JSArray
 {
     public struct Enumerator : IEnumerator<JSValue>, IEnumerator
```

Comment:
>Enumerator [](http://example.com/codeflow?start=18&length=10)

JSArray already implements all these interfaces. Should we rather change the JSArray to be a generic class?

Reply:
No. `JSArray` implements `IEnumerable<T>`, that returns an instance of this `Enumerator` struct. I didn't change this much, other than make the field name consistent with other internal `Enumerator` structs.